frontend/src/types/analytics.ts

```diff
@@ -101,6 +101,21 @@ export type Events = {
     /** The component that triggered the event */
     component: "VNoResults" | "VExternalSourceList"
   }
+  /**
+   * Description: Whenever a user changes the content type
+   * Questions:
+   *   - Which content types are most popular?
+   *   - Is there interest in the non-supported content types?
+   *   - Do users switch content types? Where in their journeys?
+   */
+  CHANGE_CONTENT_TYPE: {
+    /** The previously-set media type */
+    previous: SearchType
+    /** The new media type */
+    next: SearchType
+    /** The name of the Vue component used to switch content types. */
+    component: string
```

Comment:
I kind of wonder about using the Vue component name here rather than more stable names. We don't change component names often but the header dropdown I think has had its components changed during the redesign. In cases like that, the name of the component might change, even though the interaction area stayed the same :thinking: 

This also makes me wonder whether we should send the feature flag state in the payload for all events? If we have a feature flag that toggles a particular bit of UI or a feature on or off and it suddenly changes how a custom event trends, it would be nice to see if the feature flag preferences were changed for those events.

It isn't an issue now because I don't think we have any feature flags that would qualify for that, but something to consider for the future, especially if we use things like component names which are not necessarily stable, even if the UI area they're inside of remains stable.

On the other hand, using the component name _might_ be a good way to tell the same cases as those feature flags, but probably contingent on the implementation.

No real suggestion for a change here, just some musings about how to find stable identifiers and thing carefully about exactly what each payload item is meant to help us do. If we ask the question, why do we want to know the component name for this event, does the answer lead us to think that a different approach to this payload field would be appropriate?